scipy-stubs/signal/windows/_windows.pyi

```diff
@@ -29,36 +34,148 @@ __all__ = [
+    "tukey",
+    "tuk",
+]
+_WindowNeedsParams: TypeAlias = Literal[
+    "chebwin",
+    "cheb",
+    "dpss",
+    "gaussian",
+    "gauss",
+    "gss",
+    "general cosine",
+    "general_cosine",
+    "general gaussian",
+    "general_gaussian",
+    "general gauss",
+    "general_gauss",
+    "ggs",
+    "general hamming",
+    "general_hamming",
+    "kaiser",
+    "ksr",
+    "kaiser bessel derived",
+    "kbd",
+]
+
+def general_cosine(M: int, a: Sequence[AnyReal], sym: bool = True) -> _Array_f8_1d: ...
+def boxcar(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def triang(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def parzen(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def bohman(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def blackman(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def nuttall(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def blackmanharris(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def flattop(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def bartlett(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def hann(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def tukey(M: int, alpha: AnyReal = 0.5, sym: bool = True) -> _Array_f8_1d: ...
+def barthann(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def general_hamming(M: int, alpha: AnyReal, sym: bool = True) -> _Array_f8_1d: ...
+def hamming(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def kaiser(M: int, beta: AnyReal, sym: bool = True) -> _Array_f8_1d: ...
+def kaiser_bessel_derived(M: int, beta: AnyReal, *, sym: bool = True) -> _Array_f8_1d: ...
+def gaussian(M: int, std: AnyReal, sym: bool = True) -> _Array_f8_1d: ...
+def general_gaussian(M: int, p: AnyReal, sig: AnyReal, sym: bool = True) -> _Array_f8_1d: ...
+def chebwin(M: int, at: AnyReal, sym: bool = True) -> _Array_f8_1d: ...
+def cosine(M: int, sym: bool = True) -> _Array_f8_1d: ...
+def exponential(M: int, center: AnyReal | None = None, tau: AnyReal = 1.0, sym: bool = True) -> _Array_f8_1d: ...
+def taylor(M: int, nbar: int = 4, sll: int = 30, norm: bool = True, sym: bool = True) -> _Array_f8_1d: ...
+def lanczos(M: int, *, sym: bool = True) -> _Array_f8_1d: ...
+
+#
+@overload
 def dpss(
-    M,
-    NW,
-    Kmax: Untyped | None = None,
+    *,
+    M: int,
+    NW: AnyReal,
+    Kmax: int = ...,
     sym: bool = True,
-    norm: Untyped | None = None,
-    return_ratios: bool = False,
-) -> Untyped: ...
-def lanczos(M, *, sym: bool = True) -> Untyped: ...
-def get_window(window: Untyped, Nx: Untyped, fftbins: bool = True) -> Untyped: ...
+    norm: Literal[2, "approximate", "subsample "] | None = None,
+    return_ratios: Literal[False] = ...,
+) -> _Array_f8_2d: ...
+@overload
+def dpss(
+    *,
+    M: int,
+    NW: AnyReal,
+    Kmax: None = ...,
+    sym: bool = True,
+    norm: Literal[2, "approximate", "subsample "] | None = None,
+    return_ratios: Literal[False] = ...,
+) -> _Array_f8_1d: ...
+@overload
+def dpss(
+    *,
+    M: int,
+    NW: AnyReal,
+    Kmax: int = ...,
```

Comment:
Unlike `M`, this one looks like it accepts anything with an `__index__` method, i.e. `optype.CanIndex`.